core/kotlinx-coroutines-core/src/scheduling/CoroutineScheduler.kt

```diff
@@ -840,9 +840,17 @@ internal class CoroutineScheduler(
             }
         }
 
-        private fun doPark(nanos: Long) {
+        private fun doPark(nanos: Long): Boolean {
+            /*
+             * Here we are trying to park, then check whether there are new blocking tasks
+             * (because submitting thread could have missed this thread in tryUnpark)
+             */
             parkedWorkersStackPush(this)
+            if (!blockingQuiescence()) {
+                return false
+            }
```

Comment:
^ Code style: Format those "quick returns" without curly braces, especially since `if (!blockingQuiescence()) return` is one-liner in two other places).